internal/services/api/api_helpers.go

```diff
@@ -12,6 +13,20 @@ import (
 	"github.com/posit-dev/publisher/internal/util"
 )
 
+type jsonResponseError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+	Error   string `json:"error"`
+}
+
+func NewJsonResponseError(code int, message string, err error) jsonResponseError {
```

Comment:
Have you considered adding another error code to the mix, one that would be specific to our project and not have to be used as the REST API error code? That way we could know that a malformed configuration file with extra files is error `abc`, and we can key off of that very specifically in the UX for context-specific treatment.

Reply:
I would like to see us standardize the HTTP statuses so there are only 1-2 unique error statuses, and instead, differentiate errors by an error code field in the returned JSON.

Reply:
So, if I understand correctly, we are looking to implement something like what we have in Connect?
```go
type ErrorCode int

// Error codes for the API. These values need to remain stable, as clients may
// use them to drive error-specific behavior.
const (
    _                       = ErrorCode(iota)
    ErrorInvalidTOMLCode
    ErrorUnknownTOMLKeyCode
    ...
)

var errorMessageEntries = map[ErrorCode]struct {
    Code    int
    Message string
}{
    ErrorInvalidTOMLCode:    {http.StatusBadRequest, "This is an invalid toml."},
    ErrorUnknownTOMLKeyCode: {http.StatusBadRequest, "Unknown field present in configuration file."},
    ...
}

type jsonResponseError struct {
	ErrorCode   ErrorCode  `json:"error_code"`
	Message     string     `json:"message"`
	Error       string     `json:"error"`
}
```

Reply:
Yes, similar to that. Publisher currently has some error codes defined, but they are strings, so we might want a string value in the JSON.